src/test/java/com/rackspace/salus/telemetry/ambassador/services/AgentHistoryServiceTest.java

```diff
@@ -0,0 +1,145 @@
+import com.rackspace.salus.telemetry.repositories.AgentHistoryRepository;
+import com.rackspace.salus.test.EnableTestContainersDatabase;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import java.net.InetSocketAddress;
+import java.time.Instant;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.context.annotation.Import;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.co.jemos.podam.api.PodamFactory;
+import uk.co.jemos.podam.api.PodamFactoryImpl;
+
+@Slf4j
+@RunWith(SpringRunner.class)
+@EnableTestContainersDatabase
+@DataJpaTest(showSql = false)
+@Import({AgentHistoryService.class, DatabaseConfig.class,
+    ObjectMapper.class,
+    SimpleMeterRegistry.class
+})
+public class AgentHistoryServiceTest {
+
+  @Autowired
+  AgentHistoryRepository agentHistoryRepository;
+
+  @Autowired
+  AgentHistoryService agentHistoryService;
+
+  private AgentHistory currentAgentHistory;
+
+  private PodamFactory podamFactory = new PodamFactoryImpl();
+
+  @Mock
+  MeterRegistry meterRegistry;
+
+  @Before
+  public void setUp() {
+    AgentHistory agentHistory = new AgentHistory()
+        .setTenantId("t-1")
+        .setEnvoyId("e-1")
+        .setResourceId("r-1")
+        .setConnectedAt(Instant.now())
+        .setDisconnectedAt(Instant.now())
+        .setZoneId("z-1")
+        .setRemoteIp("0.0.0.0");
+
+    currentAgentHistory = agentHistoryRepository.save(agentHistory);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    // transactional rollback should take care of purging test data, but do a deleteAll to be sure
+    agentHistoryRepository.deleteAll();
+  }
+
+  @Test
+  public void testGetAgentHistoryForTenantAndId()  {
+    Optional<AgentHistory> agentHistoryOptional = agentHistoryService.getAgentHistoryForTenantAndEnvoyId("t-1","e-1");
+    assertExpectedAgentHistory(agentHistoryOptional);
+  }
+
+  @Test
+  public void testGetAgentHistoryForTenantAndEnvoyId()  {
+    Optional<AgentHistory> agentHistoryOptional = agentHistoryService.getAgentHistoryForTenantAndEnvoyId("t-1","e-1");
+    assertExpectedAgentHistory(agentHistoryOptional);
+  }
+
+  @Test
+  public void testGetAgentHistoryForTenantAndResource()  {
+    int pageSize = 10;
+    Pageable page = PageRequest.of(0, pageSize);
+    Page<AgentHistory> result = agentHistoryService.getAgentHistoryForTenantAndResource("t-1","r-1", page);
+    assertThat(result.getTotalElements(), equalTo(1L));
```

Comment:
This is the kind of change mentioned in https://github.com/racker/salus-telemetry-ambassador/pull/100/files#r496946445
```suggestion
    assertThat(result.getTotalElements(), equalTo(1L));
    assertExpectedAgentHistory(Optional.of(result.getContent().get(0));
```